rules/php80/tests/Rector/Switch_/ChangeSwitchToMatchRector/Fixture/skip_throwen_exception.php.inc

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Rector\Php80\Tests\Rector\Switch_\ChangeSwitchToMatchRector\Fixture;
+
+class SkipThrowenException
+{
+    public function save(): void
+    {
+        switch ('deleted') {
+            case 'inserted':
+                echo '1';
+                break;
+
+            case 'deleted':
+                throw new \Exception();
```

Comment:
Exception is an expression since PHP8. https://www.amitmerchant.com/using-throw-as-an-expression-in-php8/ So This case would be skipped because of `echo` case, NOT because of thrown exception.

Reply:
Indeed, just in this case it's a `Stmt` in php-parser nodes: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Stmt/Throw_.php

I've just noticed there is also `Expr` version: https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Expr/Throw_.php,
so the `Stmt\Throw_` would have to be converted to `Expr\Throw_`

Reply:
I think Rector could handle this, if we have failing test case.

Reply:
I think we cannot have failing test now, as rector skips case when there is an exception thrown. (even when it is possible to covert) https://getrector.org/demo/f92228fa-6001-4fcf-9fdc-35643c66bab6

Reply:
It would require to modify the rule :) do you want to try it?

Reply:
Not now, as I'm quite in hurry. I would like to fix my project before New Year. ;-)